src/main/java/sorald/SegmentRepair.java

```diff
@@ -0,0 +1,98 @@
+package sorald;
+
+import sorald.segment.Node;
+import sorald.SoraldConfig;
+import sorald.processor.SoraldAbstractProcessor;
+import spoon.Launcher;
+import spoon.processing.ProcessingManager;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.factory.Factory;
+import spoon.support.JavaOutputProcessor;
+import spoon.support.QueueProcessingManager;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+public class SegmentRepair extends SoraldAbstractRepair {
+    private HashMap<String, Integer> processorNbsRepaired = new HashMap<String, Integer>();
+
+    public SegmentRepair(SoraldConfig config) {
+        super(config);
+    }
+
+    public void repair() {
+        LinkedList<LinkedList<Node>> segments = this.config.getSegments();
+        final String outputDirPath = this.config.getWorkspace() + File.separator + Constants.SPOONED;
+
+        List<Integer> ruleKeys = this.config.getRuleKeys();
+        int ruleKey = ruleKeys.get(0);
```

Comment:
Why only the first rule?

Reply:
Hi, I did say in the previous closed PR that this mode only support single rule for now in Sorald. In Repairnator I can restart Sorald again after applying Git patches, thus it can repair multiple rule there anyways.